### PR TITLE
zos_copy mode is applied to the destination directory, a deviation from the communtiy module behavior.

### PR DESCRIPTION
**Pull request by AndreMarcel99**
##### SUMMARY
Fixed bug #619  Mode set for files is applied to destination directory by change the logic of changing files

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
The new logic is one is not a folder so only a file the mode will change not everything to be copy by the module

Captura de pantalla 2023-04-03 a la(s) 19 34 02

### CODE REVIEW

File: changelogs/fragments/619-Mode-set-for-files-is-applied-to-destination-directory.yml
Comment (rexemin) [inline changelogs/fragments/619-Mode-set-for-files-is-applied-to-destination-directory.yml:2]: Couple of typos in this fragment. I would also describe the work done on zos_copy to be a bugfix, rather than an enhancement. Something like this:
Fixed a bug where the module would change the mode for a directory when copying into it the contents of another.

**Review by rexemin — COMMENTED** [review]
Thank you so much for working on this issue. The change to zos_copy is pretty straightforward, just a couple of questions from me:
- Does the community module not change any permissions on directories? My guess is that it doesn't, so good catch on that
- You're mixing two bugfixes in this PR, did you open another PR for #417?

File: changelogs/fragments/619-Mode-set-for-files-is-applied-to-destination-directory.yml
Comment (AndreMarcel99) [reply changelogs/fragments/619-Mode-set-for-files-is-applied-to-destination-directory.yml:2]: - Ok I will make the changes in the changelogs.
- https://github.com/ansible-collections/ibm_zos_core/issues/417 is already waiting for approval I do not know why I had here the solution too, another change
- And lastly I will verify about copy folder to folder and if it's need just add another verification

**Review by ddimatos — CHANGES_REQUESTED** [review]
@AndreMarcel99 this looks good, i'll request changes since the changelog fragment has not been updated yet. 

My only other comment is if we needed a test case in place to prevent regression on this especially since this tied back to a support case, we would not want to relapse in function, although its not likely given the code changes is so small. 

At minimum at least show the output of the playbook in the issue/pr for us to see it works and some `ls` commands.

**Comment by AndreMarcel99** [comment]
The playbook as:
- hosts : zvm
  collections : 
    - ibm.ibm_zos_core
  gather_facts: no
  vars:
    ZOAU: "/zoau/v1.2.2"
    PYZ: "/python2/usr/lpp/IBM/cyp/v3r2/pyz"
    source: ./tests/
    destt: /test/
  environment: "{{ environment_vars }}"
  
  tasks:
    - name: Delete al files inside folder dest
      shell:  
        cmd: rm -r {{ destt }}
        
    - name: Create dest folder
      shell:
        cmd: mkdir {{ destt }}
      register: result
    
    - name: Adjust mode of dest folder
      shell:
        cmd: chmod 755 {{ destt }}
      register: result
    
    - name: Verify mode of destiny 
      shell:
        cmd: ls -la {{ destt }}
      register: result
    - name: Response
      debug:
        msg: "{{ result }}"
- name: Copy remote text files zos_copy.
      zos_copy:
        src: "{{ source }}"
        dest: "{{ destt }}"
        mode: 644
        force: true
        remote_src: false
    - name: Response zos_copy
      debug:
        msg: "{{ result }}"

    - name: See permisions on dest 
      shell:
        cmd: ls -la {{ destt }}
      register: result
    - name: Response permissions after
      debug:
        msg: "{{ result }}"
Captura de pantalla 2023-04-06 a la(s) 9 06 03
This is the results before the change.
After:
Captura de pantalla 2023-04-06 a la(s) 9 14 42

**Review by rexemin — APPROVED** [review]
Thanks for making the changes to the code, looks good. I reviewed the playbook you used and seems to work as intended now. IIRC there's already a testcase for permissions for files and directories.

One last thing: you could also use the [builtin `file` module](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/file_module.html) from Ansible to create (by using `state: directory`) and delete directories (`state: absent`) on a managed node.

**Comment by ddimatos** [comment]
@AndreMarcel99 thanks for sharing the playbook, @rexemin thanks for the code review as well, all looks good to me as well.

**Review by ddimatos — APPROVED** [review]
Thanks for the changes and good work @AndreMarcel99

**Pull request merged**
